content/config/completion.md

```diff
@@ -0,0 +1,96 @@
+---
+title: CLI completion
+description: Set up your shell to get autocomplete for Docker commands and flags
+keywords: cli, shell, fish, bash, zsh, completion, options
+---
+
+You can generate a shell completion script for the Docker CLI using the `docker
+completion` command. The completion script gives you word completion for
+commands, flags, and Docker objects (such as container and volume names) when
+you hit `<Tab>` as you type into your terminal.
+
+You can generate completion scripts for the following shells:
+
+- [Bash](#bash)
+- [Zsh](#zsh)
+- [fish](#fish)
+- PowerShell
+
+## Bash
+
+Bash does not support completions natively, and requires the `bash-completion` package to be installed.
+
+```bash
+# Install using APT:
+sudo apt install bash-completion
+
+# Install using Homebrew (Bash version 4 or later):
+brew install bash-completion@2
+# Homebrew install for older versions of Bash:
+brew install bash-completion
+
+# With pacman:
+sudo pacman -S bash-completion
+```
+
+After installing `bash-completion`, source the script in your shell
+configuration file (in this example, `.bashrc`):
+
+```console
+$ cat <<EOT >> ~/.bashrc
+if [ -f /etc/bash_completion ]; then
+    . /etc/bash_completion
+fi
+```
```

Comment:
This is tricky since installing with brew you need to use the `brew --prefix` command to get the path to where brew installs the binaries. So instead of `/etc/bash_completion` you would run `$(brew --prefix)/etc/bash_completion`.

Like this example here:

```console
$ cat <<EOT >> ~/.bash_profile
[[ -r "$(brew --prefix)/etc/profile.d/bash_completion.sh" ]] && . "$(brew --prefix)/etc/profile.d/bash_completion.sh"
EOT
```